src/test/java/org/broadinstitute/hellbender/utils/test/ArgumentsBuilderTest.java

```diff
@@ -30,9 +31,72 @@ public void testOneBigString(){
 
     @Test
     public void testFromArray(){
-        ArgumentsBuilder args = new ArgumentsBuilder(new Object[]{"Option=" + new File("/path/to"), "OtherOption=" + -1});
-        Assert.assertEquals(args.getArgsArray(), new String[]{"--Option","/path/to", "--OtherOption", "-1"});
+        ArgumentsBuilder args = new ArgumentsBuilder(new Object[]{"Option=" + new File("/path/to"), "OtherOption=" + -1, new File("/somewhere")});
+        Assert.assertEquals(args.getArgsArray(), new String[]{"--Option","/path/to", "--OtherOption", "-1", "/somewhere"});
     }
 
+    @Test
+    public void testAddArgument() throws Exception {
+        ArgumentsBuilder args = new ArgumentsBuilder();
+        args.addArgument("foo", "bar");
+        Assert.assertEquals(args.getArgsArray(), new String[]{"--foo","bar"});
+        Assert.assertEquals(args.getString(), "--foo bar");
+    }
+
+    @Test
+    public void testAddPositionalArgument() throws Exception {
+        ArgumentsBuilder args = new ArgumentsBuilder();
+        args.addPositionalArgument("foo");
+        args.addPositionalArgument("bar");
+        Assert.assertEquals(args.getArgsArray(), new String[]{"foo", "bar"});
+    }
+
+    @Test
+    public void testAddBooleanArgument() throws Exception {
+        ArgumentsBuilder args = new ArgumentsBuilder();
+        args.addBooleanArgument("foo", true);
+        args.addBooleanArgument("bar", false);
+        Assert.assertEquals(args.getArgsArray(), new String[]{"--foo", "true", "--bar", "false"});
+    }
+
+    @Test
+    public void testAddFileArgument() throws Exception {
+        ArgumentsBuilder args = new ArgumentsBuilder();
+        File file = new File("/path/to/somewhere");
```

Comment:
[FindBugs] INFO: DMI: Hard coded reference to an absolute pathname in org.broadinstitute.hellbender.utils.test.ArgumentsBuilderTest.testAddFileArgument()